### ISSUE TITLE
Possible regression on user rules

### ISSUE THREAD

**Issue by roadarsi** [henrypp/simplewall#1055]
Hello, @henrypp  the last update (3.5.2) broke something in the user rules when they are set to use ip range.

Defining a rule as:

- Protocol: any
- Family (any ports): any
- Direction: any
- Action: allow
- Remote rule: IP ranges 192.168.0.1-192.168.0.255
- Local rule: any
- Applications: any

Connections are always blocked.

Using the same rules but in applications choose an application for the rule, the rule works correctly.

If we set the rule to
- Protocol: any
- Family (any ports): any
- Direction: any
- Action: allow
- Remote rule: IP ranges 192.168.0.0/24
- Local rule: any
- Applications: any

The rule works correctly and connections are allowed

In the previous version (3.5.1) using IP range in the format 192.168.0.1-192.168.0-255 and applications like any the rules work normally.

I suppose some regression could have occurred.

**Comment by henrypp** [comment]
No. I tested now and there is correct filter conditions in wfp.

```xml
<item>
	<displayData>
		<name>simplewall</name>
		<description>User rule\testrange1</description>
	</displayData>
	<flags numItems="1">
		<item>FWPM_FILTER_FLAG_PERSISTENT</item>
	</flags>
	<layerKey>FWPM_LAYER_ALE_AUTH_CONNECT_V4</layerKey>
	<filterCondition numItems="1">
		<item>
			<fieldKey>FWPM_CONDITION_IP_REMOTE_ADDRESS</fieldKey>
			<matchType>FWP_MATCH_RANGE</matchType>
			<conditionValue>
				<type>FWP_RANGE_TYPE</type>
				<rangeValue>
					<valueLow>
						<type>FWP_UINT32</type>
						<uint32>192.168.4.1</uint32>
					</valueLow>
					<valueHigh>
						<type>FWP_UINT32</type>
						<uint32>192.168.4.255</uint32>
					</valueHigh>
				</rangeValue>
			</conditionValue>
		</item>
	</filterCondition>
	<action>
		<type>FWP_ACTION_PERMIT</type>
		<filterType/>
	</action>
</item>
<item>
	<displayData>
		<name>simplewall</name>
		<description>User rule\testrange2</description>
	</displayData>
	<flags numItems="1">
		<item>FWPM_FILTER_FLAG_PERSISTENT</item>
	</flags>
	<providerData/>
	<layerKey>FWPM_LAYER_ALE_AUTH_CONNECT_V4</layerKey>
	<filterCondition numItems="1">
		<item>
			<fieldKey>FWPM_CONDITION_IP_REMOTE_ADDRESS</fieldKey>
			<matchType>FWP_MATCH_EQUAL</matchType>
			<conditionValue>
				<type>FWP_V4_ADDR_MASK</type>
				<v4AddrMask>
					<addr>192.168.0.0</addr>
					<mask>255.255.255.0</mask>
				</v4AddrMask>
			</conditionValue>
		</item>
	</filterCondition>
	<action>
		<type>FWP_ACTION_PERMIT</type>
		<filterType/>
	</action>
</item>
```

**Comment by roadarsi** [comment]
Version 3.5.1- works
![image](https://user-images.githubusercontent.com/16559129/137636732-d2b568ee-b7b6-4585-bf06-577fe70b3e8e.png)

**Comment by roadarsi** [comment]
v 3.5.2
```
		<item>
			<filterKey>{c4372fc7-5e06-4adb-844f-9eda0542f0e0}</filterKey>
			<displayData>
				<name>simplewall</name>
				<description>User rule\Rede Local 2</description>
			</displayData>
			<flags numItems="1">
				<item>FWPM_FILTER_FLAG_PERSISTENT</item>
			</flags>
			<providerKey>{b0d553e2-c6a0-4a9a-aeb8-c752483ed62f}</providerKey>
			<providerData/>
			<layerKey>FWPM_LAYER_ALE_AUTH_CONNECT_V4</layerKey>
			<subLayerKey>{9fee6f59-b951-4f9a-b52f-133dcf7a4279}</subLayerKey>
			<weight>
				<type>FWP_UINT8</type>
				<uint8>11</uint8>
			</weight>
			<filterCondition numItems="2">
				<item>
					<fieldKey>FWPM_CONDITION_IP_REMOTE_ADDRESS</fieldKey>
					<matchType>FWP_MATCH_RANGE</matchType>
					<conditionValue>
						<type>FWP_RANGE_TYPE</type>
						<rangeValue>
							<valueLow>
								<type>FWP_UINT32</type>
								<uint32>192.168.0.1</uint32>
							</valueLow>
							<valueHigh>
								<type>FWP_UINT32</type>
								<uint32>192.168.0.255</uint32>
							</valueHigh>
						</rangeValue>
					</conditionValue>
				</item>
				<item>
					<fieldKey>FWPM_CONDITION_IP_REMOTE_PORT</fieldKey>
					<matchType>FWP_MATCH_EQUAL</matchType>
					<conditionValue>
						<type>FWP_UINT16</type>
						<uint16>1</uint16>
					</conditionValue>
				</item>
			</filterCondition>
			<action>
				<type>FWP_ACTION_PERMIT</type>
				<filterType/>
			</action>
			<rawContext>0</rawContext>
			<reserved/>
			<filterId>1301843</filterId>
			<effectiveWeight>
				<type>FWP_UINT64</type>
				<uint64>12790226790022905856</uint64>
			</effectiveWeight>
		</item>
```
v 3.5.1
```
		<item>
			<filterKey>{25d7ffcf-e794-4f38-b0b5-b62f3a066685}</filterKey>
			<displayData>
				<name>simplewall</name>
				<description>User rule\Rede Local 2</description>
			</displayData>
			<flags numItems="1">
				<item>FWPM_FILTER_FLAG_PERSISTENT</item>
			</flags>
			<providerKey>{b0d553e2-c6a0-4a9a-aeb8-c752483ed62f}</providerKey>
			<providerData/>
			<layerKey>FWPM_LAYER_ALE_AUTH_CONNECT_V4</layerKey>
			<subLayerKey>{9fee6f59-b951-4f9a-b52f-133dcf7a4279}</subLayerKey>
			<weight>
				<type>FWP_UINT8</type>
				<uint8>11</uint8>
			</weight>
			<filterCondition numItems="1">
				<item>
					<fieldKey>FWPM_CONDITION_IP_REMOTE_ADDRESS</fieldKey>
					<matchType>FWP_MATCH_RANGE</matchType>
					<conditionValue>
						<type>FWP_RANGE_TYPE</type>
						<rangeValue>
							<valueLow>
								<type>FWP_UINT32</type>
								<uint32>192.168.0.1</uint32>
							</valueLow>
							<valueHigh>
								<type>FWP_UINT32</type>
								<uint32>192.168.0.255</uint32>
							</valueHigh>
						</rangeValue>
					</conditionValue>
				</item>
			</filterCondition>
			<action>
				<type>FWP_ACTION_PERMIT</type>
				<filterType/>
			</action>
			<rawContext>0</rawContext>
			<reserved/>
			<filterId>1303018</filterId>
			<effectiveWeight>
				<type>FWP_UINT64</type>
				<uint64>12790222941732208640</uint64>
			</effectiveWeight>
		</item>
```

**Comment by roadarsi** [comment]
I belive this block is the cause when application is set to any, when choose a application to the rule, this block not exists and rule works on 3.5.2
```
				<item>
					<fieldKey>FWPM_CONDITION_IP_REMOTE_PORT</fieldKey>
					<matchType>FWP_MATCH_EQUAL</matchType>
					<conditionValue>
						<type>FWP_UINT16</type>
						<uint16>1</uint16>
					</conditionValue>
				</item>
```

**Comment by henrypp** [comment]
fixed in 3.5.3
